ktlint-test/src/main/kotlin/com/pinterest/ktlint/test/KtLintAssertThat.kt

```diff
@@ -652,12 +653,15 @@ internal class MissingEolMarker : RuntimeException(
  * Expectation of the [LintError]. Contrary to the [LintError] it does not contain the ruleId. The ruleId will be derived from the rule for
  * which the AssertThat was created.
  */
-public data class LintViolation(
-    val line: Int,
-    val col: Int,
-    val detail: String,
-    val canBeAutoCorrected: Boolean = true,
-)
+@Poko
+public class LintViolation
+    @JvmOverloads
```

Comment:
❗ 
This is unrelated, I just spotted, this class could use an extra interoperability with java, and I made the _default_ constructor explicitly defined, so it can be called from Java. 
This change is reflected in public api here:
<img width="583" alt="image" src="https://github.com/pinterest/ktlint/assets/36954793/2a0faab0-207c-4150-92a5-b6f18b4774b2">